clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/HubConnection.java

```diff
@@ -264,40 +266,16 @@ public HubConnectionState getConnectionState() {
                 return CompletableFuture.completedFuture(null);
             }
 
+            hubConnectionState = HubConnectionState.DISCONNECTED;
```

Comment:
I know we do this here in the Typescript client, but I think we should remove it and just rely on the `stopConnection` callback setting this.

Reply:
Yeah, the logic we used in JavaScript is because of observed browser behavior. I'd rather this be consistent with .NET (which waits for the WebSocket to close) until we see a similar behavior that we need to work around.